src/pages/signin.tsx

```diff
@@ -34,12 +39,31 @@ export default function Signin() {
   });
 
   const [serverErrorState, setServerError] = useState<string | null>(null);
+  const dispatch = useAppDispatch();
+  const currentUserid = useAppSelector(selectUserid);
+  const router = useRouter();
 
-  const onSubmitHandler: SubmitHandler<FormValues> = async () => {
+  const onSubmitHandler: SubmitHandler<FormValues> = async (formData) => {
     try {
       // @TODO Implement submit
       // await onSubmit(formData.email, formData.password);
 
+      const signInPayload = { email: formData.email, password: formData.password };
+      const signInRes = await axiosSignIn('/api/v1/auth/signin', signInPayload);
+
+      const token = signInRes.data.access_token;
+      const decoded = jwt(token);
+
+      const utcSeconds = decoded.exp;
+      const now = decoded.iat;
+      const remainingDays = Math.floor((utcSeconds - now) / 86400);
+      const decodedId = decoded.sub;
+
+      dispatch(storeuserid(decodedId));
+
+      Cookies.set('auth', '', { expires: remainingDays });
```

Comment:
All these lines can be turned into a utility function:

something in utils called `jwt.ts` that will have a `decodeToken` function `setTokenToCookie` and `getTokenFromCookie`

naming can be better, something like `expirationEpochDate, `issuedEpochDate` and `expirationDateInDays`